routes/tasks.js

```diff
@@ -155,4 +155,51 @@ router.patch('/:id', authenticate, authorizeUser('appOwner'), updateTask, tasks.
  */
 router.get('/:username', tasks.getUserTasks)
 
+/**
+ * @swagger
+ * /tasks/self/:id:
+ *  patch:
+ *   summary: used to update self task status
+ *   tags:
+ *     - Tasks
+ *   requestBody:
+ *     desciption: Task status
+ *     content:
+ *       application/json:
+ *         schema:
+ *           $ref: '#/components/schemas/tasks'
+ *   responses:
+ *     204:
+ *       description: Status of self task udpated
+ *       content:
+ *         application/json:
+ *           schma:
+ *             $ref: '#/components/schemas/tasks'
+ *     401:
+ *       description: unAuthorized
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/errors/unAuthorized'
+ *     403:
+ *       description: forbidden
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/errors/forbidden'
+ *     404:
+ *       description: notFound
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/errors/notFound'
+ *     500:
+ *       description: badImplementation
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/errors/badImplementation'
+ */
+router.patch('/self/:id', authenticate, updateTask, tasks.updateTaskStatus)
```

Comment:
We can't use the same validator `updateTask`
We need a separate validator maybe `updateSelfTask` and it should allow only status